client/src/main/scala/org/http4s/client/middleware/Retry.scala

```diff
@@ -40,14 +41,24 @@ object Retry {
           policy(req, Left(e), attempts) match {
             case Some(duration) =>
               // info instead of error(e), because e is not discarded
-              logger.info(
-                s"Request $req threw an exception on attempt #$attempts attempts. Giving up.")
+              logger.info(e)(
+                s"Request threw an exception on attempt #$attempts. Retrying after $duration")
               nextAttempt(req, attempts, duration, None)
             case None =>
+              logger.info(e)(
+                s"Request ${showRequest(req, redactHeaderWhen)} threw an exception on attempt #$attempts. Giving up."
+              ) 
               F.raiseError[DisposableResponse[F]](e)
           }
       }
 
+    def showRequest(request: Request[F], redactWhen: CaseInsensitiveString => Boolean): String = {
```

Comment:
I'd make this private so we can refactor later without breaking bincompat.

Reply:
It's actually a inner function in `apply`

Reply:
Oops. :+1: